### ISSUE TITLE
Docstring mismatch in losses.py and performance.py

### ISSUE THREAD

**Issue by MasayukiNagai** [jmschrei/bpnet-lite#26]
Hi, thanks for the update!

I noticed a potential discrepancy between the docstrings and actual return shapes in `losses.py`:

- `MNLLLoss` claims to return a scalar “averaged over all examples and all other dimensions,” but actually returns a tensor of shape (n, …) — one loss per example (and per spatial dim, if present).
- `log1pMSELoss` says it returns a (n, 1) tensor averaged over all examples, but actually returns a tensor with shape (n, ...) — one scalar per example, averaging only over the last dimension.

I suppose these are just artifacts from the previous code but just pointing it out for clarity. Thanks!
```
# MNLLLoss
	Returns
	-------
	loss: float
		The multinomial log likelihood loss of the true counts given the
		predicted probabilities, averaged over all examples and all other
		dimensions.

# log1pMSELoss
 	Returns
	-------
	loss: torch.tensor, shape=(n, 1)
		The MSE loss on the log of the two inputs, averaged over all examples
		and all other dimensions.
	"""
```

**Comment by MasayukiNagai** [comment]
Additionally, I would really appreciate it if you could update and clarify the expected input shapes for `calculate_performance_measures` function in `performance.py`. 

For example, if I assume the same shape as the docstring, due to the changes in the output shape, the output still retains the O dimension after applying `f` inside `batched_smoothed_function`. This leads to a shape mismatch between the preallocated `results[start:end]` (expected shape (B, T)) and the actual output of `f(logps_, true_counts_)` (shape (B, T, O)).

```
# logps: (N, T, O, 2)
# true_counts: (N, T, O, 2)
# pred_log_counts: (N, T)

def batched_smoothed_function(logps, true_counts, f, ...):
    n = logps.shape[0]
    results = torch.empty(*logps.shape[:2])  # shape: (N, T)

    for start in range(0, n, batch_size):
        ...
        results[start:end] = f(logps_, true_counts_)  # shape mismatch

    return results
```

In `bpnet.py`, it seems to be passing
```
y_profile shape: torch.Size([3699, 2, 1000])
y_valid shape: torch.Size([3699, 2, 1000])
y_counts shape: torch.Size([3699, 1])
```
